File: Data Structures/linear-search.c
#include <stdio.h>
int main()
{
    // Time Complexity = O(n)
    int n, count = 0, num;
    printf("Enter the size of the array: ");
    scanf("%d", &n);
    int arr[n], positions[n];
    printf("\nEnter the elements of the array: \n");
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
    }
    for (int i = 0; i < n; i++)
    {
        printf("%d, ", arr[i]);
    }
    printf("\nEnter the element you want to search: ");
    scanf("%d", &num);
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == num)
        {
            printf("\nElement is present at %d position", i + 1);
            count++;
        }
    }
    if (count == 0)
        printf("\nElement is not present in array");

    else
        printf("\n%d appears %d times in the given array", num, count);

    return 0;
}

File: DAA/linear_search.py
arr = [2,6,4,8,63,1,5,7,64,9]
key = 64
position = -1
count = 0
for x in arr:
    count+=1
    if x==key:
        count+=2
        position =  arr.index(x)
        break
count+=1
if position == -1:
    count+=2
    print("element not present in the list")
    print("Number of operations = ",count)
else:
    count+=3 
    print("element is present at ", position,"th index")
    print("Number of operations = ",count)